File: 191. Number of 1 Bits/bits.js
var hammingWeight = function (n) {
	const str = n.toString(2).match(/1/g);
	if (str) return str.length;
	else return 0;
};

const num = 00000000000000000000000000000000;
console.log(hammingWeight(num));

/*
00000000000000000000000000001011 = 3
00000000000000000000000010000000 = 1
11111111111111111111111111111101 = 31
00000000000000000000000000000000 = 0
*/

File: 191. Number of 1 Bits/bits.cpp
#include <iostream>

int hammingWeight(uint32_t n) {
  unsigned int count = 0;
  while (n) {
    count += n & 1;
    n >>= 1;
  }
  return count;
}

int main() {
  uint32_t num = 00000000000000000000000000001011;
  std::cout << hammingWeight(num);
}

/*
00000000000000000000000000001011 = 3
00000000000000000000000010000000 = 1
11111111111111111111111111111101 = 31
*/